client/forest-manager/src/rocksdb.rs

```diff
@@ -0,0 +1,611 @@
+    utils::{deserialize_value, serialize_value},
+};
+
+const LOG_TARGET: &str = "forest_storage";
+
+mod well_known_keys {
+    pub const ROOT: &[u8] = b":root";
+}
+
+pub(crate) fn other_io_error(err: String) -> io::Error {
+    io::Error::new(io::ErrorKind::Other, err)
+}
+
+/// Open the database on disk, creating it if it doesn't exist.
+fn open_creating_rocksdb(db_path: String) -> io::Result<Database> {
+    let root = PathBuf::from("/tmp/");
+    let path = root.join("storagehub").join(db_path);
+
+    let db_config = DatabaseConfig::with_columns(1);
+
+    let path_str = path
+        .to_str()
+        .ok_or_else(|| other_io_error(format!("Bad database path: {:?}", path)))?;
+
+    std::fs::create_dir_all(&path_str)?;
+    let db = Database::open(&db_config, &path_str)?;
+
+    Ok(db)
+}
+
+/// Storage backend for RocksDB.
+pub struct StorageDb<Hasher> {
+    pub db: Arc<dyn KeyValueDB>,
+    pub _phantom: std::marker::PhantomData<Hasher>,
+}
+
+impl<H: Hasher> Storage<H> for StorageDb<H> {
+    fn get(&self, key: &H::Out, prefix: Prefix) -> Result<Option<DBValue>, String> {
+        let prefixed_key = prefixed_key::<H>(key, prefix);
+        self.db.get(0, &prefixed_key).map_err(|e| {
+            warn!(target: LOG_TARGET, "Failed to read from DB: {}", e);
+            format!("Failed to read from DB: {}", e)
+        })
+    }
+}
+
+/// Backend storage used by the [`RocksDBForestStorage`].
+pub trait Backend<T: TrieLayout>: Storage<HashT<T>>
+where
+    <T::Hash as Hasher>::Out: TryFrom<Vec<u8>>,
+{
+    /// Write the transaction to the storage.
+    fn write(&mut self, transaction: DBTransaction) -> Result<(), ForestStorageErrors>;
+    /// Get the [`ROOT`](`well_known_keys::ROOT`) from storage.
+    fn storage_root(&self) -> Result<Option<HasherOutT<T>>, ForestStorageErrors>;
+}
+
+impl<T: TrieLayout> Backend<T> for StorageDb<HashT<T>>
+where
+    <T::Hash as Hasher>::Out: TryFrom<Vec<u8>>,
+{
+    fn write(&mut self, transaction: DBTransaction) -> Result<(), ForestStorageErrors> {
+        self.db.write(transaction).map_err(|e| {
+            warn!(target: LOG_TARGET, "Failed to write to DB: {}", e);
+            ForestStorageErrors::FailedToWriteToStorage
+        })
+    }
+
+    fn storage_root(&self) -> Result<Option<HasherOutT<T>>, ForestStorageErrors> {
+        let maybe_root = self.db.get(0, well_known_keys::ROOT).map_err(|e| {
+            warn!(target: LOG_TARGET, "Failed to read root from DB: {}", e);
+            ForestStorageErrors::FailedToReadStorage
+        })?;
+
+        let root = maybe_root
+            .map(|root| {
+                HasherOutT::<T>::try_from(root).map_err(|_| {
+                    warn!(target: LOG_TARGET, "Failed to parse root from DB");
+                    ForestStorageErrors::FailedToParseRoot
+                })
+            })
+            .transpose()?;
+
+        Ok(root)
+    }
+}
+
+/// Forest storage backend using RocksDB.
```

Comment:
Confusing docs. `StorageDb` is commented to be the "Storage backend for RocksDB", and now this is the "Forest storage backend using RocksDB"? At this point I would ask what is a storage backend? 😅

Reply:
What about `/// RocksDB based [`ForestStorage`] implementation.`?